runner/debug/consul.go

```diff
@@ -0,0 +1,101 @@
+package debug
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/hcdiag/op"
+	"github.com/hashicorp/hcdiag/redact"
+	"github.com/hashicorp/hcdiag/runner"
+)
+
+var _ runner.Runner = ConsulDebug{}
+
+// ConsulDebugConfig is a config struct for ConsulDebug runners
+type ConsulDebugConfig struct {
+	Archive    string
+	Duration   string
+	Interval   string
+	Captures   []string
+	Redactions []*redact.Redact
+}
+
+// ConsulDebug represents a ConsulDebug runner
+type ConsulDebug struct {
+	Archive    string           `json:"archive"`
+	Duration   string           `json:"duration"`
+	Interval   string           `json:"interval"`
+	Captures   []string         `json:"captures"`
+	Redactions []*redact.Redact `json:"redactions"`
+
+	output string
+}
+
+func (d ConsulDebug) ID() string {
+	return "ConsulDebug"
+}
+
+func NewConsulDebug(cfg ConsulDebugConfig, tmpDir string, debugDuration time.Duration, debugInterval time.Duration) *ConsulDebug {
+	// Create a pseudorandom string of characters to allow >1 ConsulDebug runner without filename collisions
```

Comment:
Is this something that could be done with `os.MkdirTemp` instead, similar to what we do with agent.CreateTemp?

Reply:
When I looked at the `randAlphanumString` function, I noticed that `alphaNumBytes` is missing the number 7; I can't add a comment since that wasn't changed in this PR, but if we use that function still, would you mind taking care of that too?

Reply:
Woah, nice! I had no idea os.MkdirTemp can add random patterns -- I'll definitely use that.